src/battle_ai_main.c

```diff
@@ -3217,6 +3228,59 @@ static s32 AI_CompareDamagingMoves(u32 battlerAtk, u32 battlerDef, u32 currId)
         }
     }
 
+    // Compare damage rather than hits to KO for AI_FLAG_RISKY   
+    if ((AI_THINKING_STRUCT->aiFlags[battlerAtk] & AI_FLAG_RISKY) && bestDamage == damage[currId])
```

Comment:
How likely is it that more then one move will have the same damage while being the best? I wonder if it makes sense to just increase score for best damage and ignore further calcs?

Reply:
I think it'd be super unlikely and I'm open to this approach for sure, I just figured I'd leave in all the niche-case code from the main approach for the sake of accuracy.

D'you think having worse niche-case behaviour is worth the cleaner code? The niche cases are gonna be pretty few and far between.

Reply:
For AI I prefer cleaner code in favor of addressing very niche cases because the AI code is already huge.